scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -105,6 +118,32 @@ public Connection(final SessionTicket sessionTicket, SessionId sessionId) {
 		this.peerAddress = null;
 		this.autoResumptionTimeout = null;
 		this.sessionListener = null;
+		this.maxDeferredProcessedApplicationDataMessages = 0;
+		this.deferredApplicationData = EMPTY;
+	}
+
+	public void addApplicationDataForDeferredProcessing(RawData outgoingMessage) {
+		// backwards compatibility, allow on outgoing message to be stored.
+		int max = maxDeferredProcessedApplicationDataMessages == 0 ? 1 : maxDeferredProcessedApplicationDataMessages;
```

Comment:
Set it to 0 means, the one outgoing (as the implementation  before) is stored and no incoming.
As the comment says, it's only for the case someone want the old behaviour.

Reply:
A `1` would change the behaviour for incoming handshakes and application data.
Therefore, even if I think, it will not be used too often, I implemented the 0 as backwards compatible.